### ISSUE TITLE
`orm.load_only`, `orm.defer` and `orm.undefer` wrongfully typed with `str | QueryableAttribute`

### ISSUE THREAD

**Issue by provinzkraut** [sqlalchemy/sqlalchemy#10131]
### Ensure stubs packages are not installed

- [X] No sqlalchemy stub packages is installed (both `sqlalchemy-stubs` and `sqlalchemy2-stubs` are not compatible with v2)

### Verify if the api is typed

- [X] The api is not in a module listed in [#6810](https://github.com/sqlalchemy/sqlalchemy/issues/6810) so it should pass type checking

### Describe the typing issue

The functions `orm.load_only`, `orm.defer` and `orm.undefer` are typed as accepting `key: str | QueryableAttribute` as their first argument(s), but will raise an `sqlalchemy.exc.ArgumentError` when strings are passed.

### To Reproduce

```python
from sqlalchemy.orm import load_only, defer, undefer


load_only("foo")
defer("foo")
undefer("foo")
```

### Error

```
Traceback (most recent call last):
  File "/home/***/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 2272, in _parse_attr_argument
    insp: InspectionAttr = inspect(attr)  # type: ignore
  File "/home/***/python3.10/site-packages/sqlalchemy/inspection.py", line 152, in inspect
    raise exc.NoInspectionAvailable(
sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <class 'str'>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/***/test.py", line 4, in <module>
    load_only("foo")
  File "/home/***/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 2325, in load_only
    _, lead_element, _ = _parse_attr_argument(attrs[0])
  File "/home/***/python3.10/site-packages/sqlalchemy/orm/strategy_options.py", line 2274, in _parse_attr_argument
    raise sa_exc.ArgumentError(
sqlalchemy.exc.ArgumentError: expected ORM mapped attribute for loader strategy argument

``` 

### Versions

- OS:
- Python: 3.10.10
- SQLAlchemy: 2.0.19
- Type checker (eg: mypy 0.991, pyright 1.1.290, etc): N/A

**Comment by CaselIT** [comment]
Hi,

thanks for reporting, that string should probably be a `literal`

**Comment by provinzkraut** [comment]
What would that literal be?

**Comment by CaselIT** [comment]
`'*'` may still work, but I would need to check.

btw it seems that the documentation is duplicated for some reason https://docs.sqlalchemy.org/en/20/orm/queryguide/columns.html#sqlalchemy.orm.load_only

**Comment by provinzkraut** [comment]
Thanks for the pointers! I'll check this out and could submit a PR tomorrow

**Comment by CaselIT** [comment]
thanks!

**Comment by sqla-tester** [comment]
**Janek Nouvertné** has proposed a fix for this issue in the **main** branch:

**fix #10131: update type annotations for `load_only`, `defer` and `undefer`** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4826

**Comment by sqla-tester** [comment]
**Janek Nouvertné** has proposed a fix for this issue in the **main** branch:

**Update type annotations for loading options** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4826
